src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/yflow/update/action/ValidateYFlowAction.java

```diff
@@ -102,6 +102,9 @@ public ValidateYFlowAction(PersistenceManager persistenceManager, FlowOperations
                     format("Provided sub-flow ids is not valid for the y-flow %s", yFlowId));
         }
 
+        String mainAffinityFlowId = yFlow.getSubFlows().iterator().next().getFlow().getAffinityGroupId();
```

Comment:
I'd replace "yFlow.getSubFlows().iterator().next()" with yFlow.getSubFlows().stream().findFirst().orElseThrow(...) to validate the y-flow consistency.

Reply:
Fixed at your suggestion.